Assets/Shaders/SplatCube.shader

```diff
@@ -0,0 +1,207 @@
+            res.cavity = lerp(_mat1.cavity, _mat2.cavity, _weight);
+
+            return res;
+        }
+
+        /// Samples current material using nearest filtering
+        MaterialSample SampleNearestMaterial(in float2 _uv, in float2 _uvDetail)
+        {
+            const uint matIndex = FetchMaterialIndex(_uv);
+            return FetchMaterialValues(matIndex, _uvDetail);
+        }
+
+        // Samples current material using bilinear filtering
+        MaterialSample SampleBilinearMaterial(in float2 _uv, in float2 _uvDetail)
+        {
+            const float2 texelSize = _SplatMap_TexelSize.xy;
+            const float2 textureDims = _SplatMap_TexelSize.zw;
+
+            const float2 floatIuv = _uv * textureDims - float2(0.5, 0.5);
+            const float2 iuv = floor(floatIuv) * texelSize;
+
+            // manually compute the weight to perform bilinear filtering
+            const float coeffX = frac(floatIuv.x);
+            const float coeffY =  frac(floatIuv.y);
+
+            const uint matIndex00 = FetchMaterialIndex(iuv + float2(          0,           0));
+            const uint matIndex01 = FetchMaterialIndex(iuv + float2(          0, texelSize.y));
+            const uint matIndex10 = FetchMaterialIndex(iuv + float2(texelSize.x,           0));
+            const uint matIndex11 = FetchMaterialIndex(iuv + float2(texelSize.x, texelSize.y));
+
+            // early discard when dealing with a single material
+            if ((matIndex00 == matIndex10) && (matIndex01 == matIndex11) && (matIndex00 == matIndex01))
+            {
+                return FetchMaterialValues(matIndex00, _uvDetail);
+            }
+
+            MaterialSample mat00 = FetchMaterialValues(matIndex00, _uvDetail);
+
+            if (matIndex00 != matIndex10)
+            {
+                const MaterialSample mat10 = FetchMaterialValues(matIndex10, _uvDetail);
+                mat00 = LerpMaterial(mat00, mat10, coeffX);
+            }
+
+            MaterialSample mat01 = FetchMaterialValues(matIndex01, _uvDetail);
+
+            if (matIndex01 != matIndex11)
+            {
+                const MaterialSample mat11 = FetchMaterialValues(matIndex11, _uvDetail);
+                mat01 = LerpMaterial(mat01, mat11, coeffX);
+            }
+
+            return LerpMaterial(mat00, mat01, coeffY);
+        }
+
+        // Randomizes given uvs to introduce noise on material transitions
+        float2 RandomizeUv(in float2 _baseUv, in float _noiseScale, in float _noiseRadius)
+        {
+            const float2 texelSize = _SplatMap_TexelSize.xy;
+            const float2 noiseUv = _noiseScale * _baseUv;
+
+            const float noiseA = tex2D(_NoiseMap, noiseUv).r;
+            const float noiseB = tex2D(_NoiseMap, noiseUv + float2(0.5, 0.5)).r;
+
+            float2 offset = float2(noiseA, noiseB);
+            offset = 2 * offset - 1;
+
+            return clamp(_baseUv + _noiseRadius * texelSize * offset, 0.0, 1.0);
+        }
```

Comment:
I will remove this method, as well the noise parameters and textures because we don't really need it and it wouldn't work well on a non-planar terrain anyway.

Reply:
https://github.com/davidasberg/cubes_traveller/pull/11/commits/08b629e7dd2ecaeffb23cb3b893479de655ce7ed